File: Tests/TypeSharp.Test/SuperTest.ts
declare namespace TypeSharp.Test {
    interface SuperClass {
        nested?: SuperClass.NestedClass;
        overrideInt?: number;
        newObject?: any;
    }
    interface ChildClass {
        overrideInt?: number;
        newObject?: SuperClass.NestedClass;
        nested?: SuperClass.NestedClass;
    }
}
declare namespace TypeSharp.Test.SuperClass {
    interface NestedClass {
        intValue?: number;
    }
}

File: Tests/TypeSharp.Test/SuperTest.cs
using NStandard;
using System.Reflection;
using TypeSharp.Test.Models;
using Xunit;

namespace TypeSharp.Test
{
    public partial class CommonTests
    {
        [Fact]
        public void SuperTest()
        {
            var types = Assembly.GetExecutingAssembly().GetTypesWhichMarkedAs<TypeScriptModelAttribute>();
            var builder = new TypeScriptModelBuilder();
            builder.CacheType<SuperClass>();
            builder.CacheType<ChildClass>();

            var tscode = builder.Compile();
            Util.Assert(tscode);
        }
    }
}
